src/shared/components/DatePicker/DatePicker.tsx

```diff
@@ -150,7 +150,9 @@ export const DatePicker = <T extends FieldValues>({
                 id={id}
                 open={isOpen}
                 anchorEl={anchorEl}
-                onClose={handlePickerClose}
+                onClose={() => {
+                  handlePickerClose(getSelectedDate());
```

Comment:
could you please move the logic into a separate function to use it here and in row 195, for ex.:
```
onClose={handlePickerCloseWithSelectedDate}
...
onClick={handleCloseWithSelectedDate}
```